BotProject/Templates/CSharp/Scripts/Node/botProjectDeploy.ts

```diff
@@ -0,0 +1,506 @@
+            fs.writeFile(logFile, JSON.stringify(data, null, 4), (err) => {
+                if (err) {
+                    reject(err)
+                }
+                resolve()
+            })
+        })
+    }
+
+    private async getFiles(dir: string): Promise<string[]> {
+        const dirents = await readdir(dir, { withFileTypes: true });
+        const files = await Promise.all(dirents.map((dirent) => {
+            const res = path.resolve(dir, dirent.name);
+            return dirent.isDirectory() ? this.getFiles(res) : res;
+        }));
+        return Array.prototype.concat(...files);
+    }
+
+    private async botPrepareDeploy(pathToDeploymentFile: string) {
+        return new Promise((resolve, reject) => {
+            const data = `[config]\nproject = BotProject.csproj`
+            fs.writeFile(pathToDeploymentFile, data, (err) => {
+                reject(err)
+            })
+        })
+    }
+
+    private async dotnetPublish(publishFolder: string, projFolder: string, botPath?: string) {
+        const projectPath = path.join(projFolder, 'BotProject.csproj')
+        await exec(`dotnet publish ${projectPath} -c release -o ${publishFolder} -v q`)
+        return new Promise((resolve, reject) => {
+            const remoteBotPath = path.join(publishFolder, 'ComposerDialogs')
+            const localBotPath = path.join(projFolder, 'ComposerDialogs')
+
+            if (botPath) {
+                console.log(`Publishing dialogs from external bot project: ${botPath}`)
+                fs.copy(botPath, remoteBotPath, {
+                    overwrite: true,
+                    recursive: true
+                }, (err) => {
+                    reject(err)
+                })
+            }
+            else {
+                fs.copy(localBotPath, remoteBotPath, {
+                    overwrite: true,
+                    recursive: true
+                }, (err) => {
+                    reject(err)
+                })
+            }
+            resolve()
+        })
+    }
+
+    private async zipDirectory(source: string, out: string) {
+        const archive = archiver('zip', { zlib: { level: 9 } });
+        const stream = fs.createWriteStream(out);
+
+        return new Promise((resolve, reject) => {
+            archive
+                .directory(source, false)
+                .on('error', err => reject(err))
+                .pipe(stream)
+                ;
+
+            stream.on('close', () => resolve());
+            archive.finalize();
+        });
+    }
+
+    private notEmptyLuisModel(file: string) {
+        return fs.readFileSync(file).length > 0
+    }
+
+    public async deploy(name: string, environment: string, luisAuthoringKey?: string, luisAuthoringRegion?: string, logFile?: string, botPath?: string, language?: string) {
+        if (!logFile) {
+            logFile = 'deploy_log.txt'
+        }
+        if (!this.creds) {
+            this.creds = await msRestNodeAuth.interactiveLogin()
```

Comment:
can this interactiveLogin been invoked in non-cli env? replace this with a token parameter?